include/translated-object.php

```diff
@@ -377,92 +283,48 @@ public function get_translation( $id, $lang ) {
 	 *
 	 * @since 0.1
+	 * @since 3.4 Returns 0 instead of false.
+	 *
+	 * @param int                     $id   Object ID.
+	 * @param PLL_Language|string|int $lang Language (object, slug, or term ID).
+	 * @return int The translation object ID if exists, otherwise the passed ID. `0` if the passed object has no language.
 	 *
-	 * @param int                     $id   Object id ( typically a post_id or term_id ).
-	 * @param int|string|PLL_Language $lang Language ( term_id or slug or object ).
-	 * @return int|false The translation object id if exists, otherwise the passed id, false if the passed object has no language.
+	 * @phpstan-return int<0, max>
 	 */
 	public function get( $id, $lang ) {
 		$id = $this->sanitize_int_id( $id );
 
 		if ( empty( $id ) ) {
-			return false;
+			return 0;
 		}
 
 		$lang = $this->model->get_language( $lang );
 
 		if ( empty( $lang ) ) {
-			return false;
+			return 0;
 		}
 
 		$obj_lang = $this->get_language( $id );
 
 		if ( empty( $obj_lang ) ) {
-			return false;
+			return 0;
 		}
 
-		return $obj_lang->term_id === $lang->term_id ? $id : $this->get_translation( $id, $lang );
-	}
-
-	/**
-	 * A join clause to add to sql queries when filtering by language is needed directly in query.
-	 *
-	 * @since 1.2
-	 *
-	 * @param string $alias Optional alias for object table.
-	 * @return string Join clause.
-	 */
-	abstract public function join_clause( $alias = '' );
-
-	/**
-	 * A where clause to add to sql queries when filtering by language is needed directly in query.
-	 *
-	 * @since 1.2
-	 *
-	 * @param PLL_Language|PLL_Language[]|string|string[] $lang PLL_Language object or a comma separated list of language slug or an array of language slugs or objects.
-	 * @return string Where clause.
-	 */
-	public function where_clause( $lang ) {
-		$tt_id = $this->tax_tt;
+		$lang_term_id     = $lang->get_tax_prop( $this->get_tax_language(), 'term_id' );
+		$obj_lang_term_id = $obj_lang->get_tax_prop( $this->get_tax_language(), 'term_id' );
 
-		/*
-		 * $lang is an object.
-		 * This is generally the case if the query is coming from Polylang.
-		 */
-		if ( is_object( $lang ) ) {
-			return ' AND pll_tr.term_taxonomy_id = ' . absint( $lang->$tt_id );
-		}
-
-		/*
-		 * $lang is an array of objects, an array of slugs, or a comma separated list of slugs.
-		 * The comma separated list of slugs can happen if the query is coming from outside with a 'lang' parameter.
-		 */
-		$languages        = is_array( $lang ) ? $lang : explode( ',', $lang );
-		$languages_tt_ids = array();
-		foreach ( $languages as $language ) {
-			$language = $this->model->get_language( $language );
-
-			if ( ! empty( $language ) ) {
-				$languages_tt_ids[] = absint( $language->$tt_id );
-			}
-		}
-
-		if ( empty( $languages_tt_ids ) ) {
-			return '';
-		}
-
-		return ' AND pll_tr.term_taxonomy_id IN ( ' . implode( ',', $languages_tt_ids ) . ' )';
+		return $lang_term_id === $obj_lang_term_id ? $id : (int) $this->get_translation( $id, $lang );
```

Comment:
Since we keep the public property `$term_id` in the end, [d7432cd](https://github.com/polylang/polylang/pull/1159/commits/d7432cd1bf9b8882c44a2a98eb53f54474948ef1) this part to use it directly.